### PR TITLE
feat(@angular-devkit/build-ng-packagr): implement stable architect API

### PR DESCRIPTION
**Pull request by clydin**
This is a continuation of the Angular builder refactoring following the introduction of the new stable architect builder API.

### CODE REVIEW

File: packages/angular_devkit/build_ng_packagr/src/build/schema.json [hunk 1]
```diff
@@ -1,4 +1,5 @@
 {
+  "$schema": "http://json-schema.org/draft-07/schema",
```
Comment (alan-agius4) [inline packages/angular_devkit/build_ng_packagr/src/build/schema.json]: We we also update 
```
  "project": {
      "type": "string",
      "description": "The file path of the package.json for distribution via npm."
    },
```

to 
```
  "project": {
      "type": "string",
      "description": "The full path for the ng-packagr configuration file, relative to the current workspace."
    },
```

File: packages/angular_devkit/build_ng_packagr/src/build/index2.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  root: string,
+): Promise<import ('ng-packagr').NgPackagr> {
+  const packager = (await import('ng-packagr')).ngPackagr();
+
+  packager.forProject(resolve(root, options.project));
+
+  if (options.tsConfig) {
+    packager.withTsConfig(resolve(root, options.tsConfig));
+  }
+
```
Comment (alan-agius4) [inline packages/angular_devkit/build_ng_packagr/src/build/index2.ts]: You can do something
```
packager.withOptions({
		watch: options.watch
})
```

File: packages/angular_devkit/build_ng_packagr/src/build/index2.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  return packager;
+}
+
+export function buildLibrary(
+  options: NgPackagrBuilderOptions,
+  context: BuilderContext,
+): Observable<BuilderOutput> {
+  return from(initialize(options, context.workspaceRoot)).pipe(
+    switchMap(packager => options.watch ? packager.watch() : packager.build()),
```
Comment (alan-agius4) [inline packages/angular_devkit/build_ng_packagr/src/build/index2.ts]: Nit
```ts
switchMap(packager => packagr.buildAsObservable())
```

File: packages/angular_devkit/build_ng_packagr/src/build/schema.json (hunk 1 above)
Comment (hansl) [inline packages/angular_devkit/build_ng_packagr/src/build/schema.json]: Considering the draft won't be changing much, we should start using a non-draft version of the schema spec; `http://json-schema.org/schema`.

Comment (clydin) [reply packages/angular_devkit/build_ng_packagr/src/build/schema.json]: `http://json-schema.org/schema` points to latest version of the schema and will change over time.  It currently points to the `draft-07` version which allows everything to work for now.  However, there is no guarantee that the latest version will be supported by the CLI nor that the file will conform to the latest version.  By specifying a specific version the `$schema` identifier can be used as both a version identifier for the format of the file and as the location of the schema.  This allows processors to use the specified version/format of the specification when parsing and processing the file.  (ref: http://json-schema.org/draft-07/json-schema-core.html#rfc.section.7)

File: packages/angular_devkit/build_ng_packagr/src/build/index2.ts (hunk 3 above)
Comment (clydin) [reply packages/angular_devkit/build_ng_packagr/src/build/index2.ts]: Will `watch` and `build` be available longterm?  I think I actually prefer the separate methods as they are really different actions; and `build` returning a promise helps simplify and demonstrate that a build is a one time event.

Comment (alan-agius4) [reply packages/angular_devkit/build_ng_packagr/src/build/index2.ts]: yeah should be.

File: packages/angular_devkit/build_ng_packagr/src/build/schema.json (hunk 1 above)
Comment (hansl) [reply packages/angular_devkit/build_ng_packagr/src/build/schema.json]: (we use that value too when running quicktype)

This is non blocking, but we use the "latest" URI in all but 2 of our schemas, so I'd say it's better to use the same everywhere than just keep splitting the code base.

**Pull request merged**